src/porepy/numerics/time_step_control.py

```diff
@@ -1,130 +1,191 @@
+            the first and second elements of the tuple corresponds to the minimum and
+            maximum time steps, respectively.
+        iter_max: Maximum number of iterations.
+        iter_optimal_range: Optimal iteration range. The first and second elements of the
+            tuple corresponds to the lower and upper bounds of the optimal iteration range.
+        iter_lowupp_factor: Lower and upper multiplication factors. The first and second
+            elements of the tuple corresponds to the lower and upper multiplication
+            factors, respectively. We require the lower multiplication factor to be
+            strictly greater than one, whereas the upper multiplication factor is
+            required to be strictly less than one.
+        recomp_factor: Failed-to-converge recomputation factor. Factor by which the
+            time step will be multiplied in case the solution must be recomputed (see
+            documentation of `next_time_step` method).
+        recomp_max: Failed-to-converge maximum recomputation attempts. The maximum
+            allowable number of consecutive recomputation attempts. If `recomp_max` is
+            exceeded, an error will be raised.
+        print_info. Print time-stepping information.
+
+    Example:
+        # The following is an example on how to construct a time-stepping object
+        tsc = pp.TimeSteppingControl(
+            schedule=[0, 10],
+            dt_init=0.5,
+            dt_min_max=(0.1, 2),
+            iter_max=10,
+            iter_optimal_range=(4, 7),
+            iter_lowupp_factor=(1.1, 0.9),
+            recomp_factor=0.5,
+            recomp_max=5,
+            print_info=True
+        )
+        # To inspect the current attributes of the object
+        print(tsc)
+
+    Attributes:
+        dt (float): Time step.
+        dt_init (float): Intial time step.
+        dt_max (float): Maximum time step.
+        dt_min (float): Minimum time step.
+        is_constant (bool): Constant time step flag.
+        iter_low (int): Lower bound of the optimal iteration range.
+        iter_low_factor (float): Lower multiplication factor. Strictly greater than one.
+        iter_max (int): Maximum number of iterations.
+        iter_upp (int): Upper bound of the optimal iteration range.
+        iter_upp_factor (float): Upper multiplication factor. Strictly lower than one.
+        recomp_factor (float): Recomputation factor. Strictly lower than one.
+        recomp_max (int): Maximum number of recomputation attempts.
+        schedule (list): List of scheduled times including initial and final times.
+        time (float): Current time.
+        time_final (float): Final simulation time.
+        time_init (float): Initial simulation time.
+    """
 
     def __init__(
         self,
         schedule: list,
         dt_init: float,
-        dt_min_max: Tuple[float, float],
+        constant_dt: bool = False,
+        dt_min_max: Optional[Tuple[float, float]] = None,
```

Comment:
```suggestion
        dt_min_max: Optional[tuple[float, float]] = None,
```
Please also update in other parts of this module.